src/main.rs

```diff
@@ -1,35 +1,690 @@
+    /// The `fill(t: Option<T>)` method accepts an [`std::option::Option`] and returns a `GiftBox`
+    /// with either `GiftBox::Gifts(t)` if `Some(t)` was provided or a `GiftBox::Empty` if
+    /// `None` is provided.
+    ///
+    /// #Arguments
+    /// * `t: Option<T>` - `t` accepts an [`std::option::Option`]
+    ///
+    /// #Returns
+    /// Returns a `GiftBox<T>`.
+    ///
+    /// #Example
+    /// Filling a `GiftBox` with `Some(T)`:
+    /// ```
+    /// let filled_box = GiftBox::fill(Some(["Toys", "Candy", "Money"]));
+    /// assert_eq!(filled_box, GiftBox::Gifts(["Toys", "Candy", "Money"]));
+    /// ```
+    /// Filling a `GiftBox` with `None`:
+    /// ```
+    /// let empty_box: GiftBox<()> = GiftBox::fill(None);
+    ///  assert_eq!(empty_box, GiftBox::Empty);
+    /// ```
+    pub fn fill(t: Option<T>) -> GiftBox<T> {
+        match t {
+            Some(t) => GiftBox::Gifts(t),
+            None => GiftBox::Empty,
+        }
+    }
+
+    /// The `fill_keeping_some(t: Option<T>)` method accepts an
+    /// [`std::option::Option`] and returns a `GiftBox` with either `GiftBox::Gifts(Some(t))` if
+    /// `Some(t)` was provided or a `GiftBox::Empty` if `None` is provided.
+    ///
+    /// #Arguments
+    /// * `t: Option<T>` - `t` accepts an [`std::option::Option`]
+    ///
+    /// #Returns
+    /// Returns a `GiftBox<Option<T>>`.
+    ///
+    /// #Example
+    /// Filling a `GiftBox` with `Some(T)`:
+    /// ```
+    /// let filled_box = GiftBox::fill_keeping_some(Some(["Toys", "Candy", "Money"]));
+    /// assert_eq!(filled_box, GiftBox::Gifts(Some(["Toys", "Candy", "Money"])));
+    /// ```
+    /// Filling a `GiftBox` with `None`:
+    /// ```
+    /// let empty_box: GiftBox<()> = GiftBox::fill_keeping_some(None);
+    ///  assert_eq!(empty_box, GiftBox::Empty);
+    /// ```
+    pub fn fill_keeping_some(t: Option<T>) -> GiftBox<Option<T>>{
+        match t {
+            Some(t) => GiftBox::Gifts(Some(t)),
+            None => GiftBox::Empty,
+        }
+    }
 
-    // Attempt to `println!` to see what the gift box contains. However, this will not compile! An
-    // error will be displayed stating "cannot infer type for type parameter `T` declared on the
-    // enum `GiftBox`".
-    println!("The gift box contains: {:?}", empty_box);
+    /// The `open()` method takes a GiftBox and returns the contents of that GiftBox.
+    ///
+    /// #Arguments
+    /// * `self` only.
+    ///
+    /// #Returns
+    /// Returns `T` where `T` is the contents of a `GiftBox:Gifts(<T>)`.
+    ///
+    /// #Panics!
+    /// The `open()` method will panic if used on a GiftBox that is empty (`GiftBox::Empty`).
+    ///
+    /// #Example
+    /// ```
+    ///  let filled_box = GiftBox::fill(Some(vec![1, 3, 5, 7]));
+    ///  assert_eq!(filled_box.open(), vec![1, 3, 5, 7]);
+    /// ```
+    pub fn open(self) -> T {
+        match self {
+            Gifts(gift) => gift,
+            Empty => panic!("Opened an empty gift box!"),
```

Comment:
As a note. I did put this in my example just as something I could write quickly. In most idiomatic rust though, you want to avoid panicing unless something happened which as a programmer you think should be impossible, but the compiler just can tell that. The idiomatic thing to do would return something like a  `Result<T, String>` which will capture the error if something goes wrong, but still pass the value if things are as expected.

That said, Result itself does have a function `unwrap()` which panics if you call it on an error, so its not like its wrong, just something to keep in mind.

Reply:
Should I remove all panics from GiftBox as it is? I've included the panics as a nod to how Option behaves.